packages/block-editor/src/utils/dom.js

```diff
@@ -17,16 +18,28 @@ export function isInSameBlock( a, b ) {
  * Returns true if an element is considered part of the block and not its inner
  * blocks or appender.
  *
- * @param {Element} blockElement Block container element.
- * @param {Element} element      Element.
+ * @param {Element}                        blockElement Block container element.
+ * @param {Element}                        element      Element.
+ * @param {{ skipBlockInserter: boolean }} options      Custom options to set.
  *
  * @return {boolean} Whether an element is considered part of the block and not
  *                   its inner blocks or appender.
  */
-export function isInsideRootBlock( blockElement, element ) {
+export function isInsideRootBlock(
+	blockElement,
+	element,
+	options = { skipBlockInserter: false }
+) {
```

Comment:
> To be honest, using the appender classnames is probably not very reliable, as blocks can use custom appenders that may not even have a classname.

The problem is, very little thought was put in to inner blocks from an accessibility perspective. Having all these child divs and elements inside parent block elements has made it a nightmare to work with when it comes down to controlling what screen readers see. I would have much rather represented inner blocks with CSS positioning than quite literally in the DOM. Even adding things such as aria-description can be super problematic because the descriptions are read from inner element to outer element instead of just for the current element. Some type of conditional checking would have also been nice that way I could detect if I was in parent block vs. child block, but as you can see, it's all kind of hacking at the end of the day.

As far as the classes are concerned, it goes back to my previous point. I have no idea how to actually get around it. I don't think it's in the store anywhere and I don't think any custom reliable attributes are outputted. My goal was to make this work with blocks we control and then figure out the rest later.

I'm getting at, previous design choices have made things really tough for us now.

As far as what the function should do.
1. Compare the block wrapper and element to see if it is in the block.
2. Ensure the element is not in an inner block.
3. Return false if appender. The reason being is the group block is already technically an inner block but if I excluded inner blocks even if it was the parent block, focus would not work. Take the columns block for example. You have columns, column, and inner blocks of column. If I select say a image block in column 1, this is technically an inner block that needs to be focused.

Sorry this is so hard to explain. 😞 We can't change what happened before, just need to improve it going forward. 👍

Reply:
@talldan I was unsure if your suggestion would work as I almost took that approach at first. Seems to work fine. Cleaned up the code and committed.

I also didn't know if this was used elsewhere so I decided the option approach was safer. If you think it's fine, I removed it. Can always revert that commit if someone else feels differently.

I don't have enough history to know if it's safe or not... 🙂